apps/rush-lib/src/utilities/AsyncRecycler.ts

```diff
@@ -73,7 +73,25 @@ export class AsyncRecycler {
     }
 
     Utilities.retryUntilTimeout(
-      () => FileSystem.move({ sourcePath: folderPath, destinationPath: newFolderPath }),
+      () => {
+        try {
+          FileSystem.move({ sourcePath: folderPath, destinationPath: newFolderPath });
+        } catch (e) {
+          if (FileSystem.isErrnoException(e)) {
+            // It could be an invalid symlink, so try to obtain the stats
```

Comment:
How would this arise?

Reply:
It was coming up when PNPM was stopped from performing its install midway through, leaving the symlink in a bad state. However, I decided to revert this functionality and rely on PNPM to clean up individual project node_modules folders, as it appears to handle it correctly. For now, we will continue doing what we normally do (delete the node_modules folder in common/temp)

Reply:
👍 That's a better approach